packages/app/src/systems/Account/components/FuelAddress/FuelAddress.tsx

```diff
@@ -11,35 +10,19 @@ export type AddressProps = {
 };
 
 export const FuelAddress = ({ address, css }: AddressProps) => {
-  const isValidAddress = isB256(address) || isBech32(address);
-  const [showB256, setShowB56] = useState(false);
-  const getAddress = () => {
-    const addressInstance = Address.fromString(address);
-    return showB256 ? addressInstance.toB256() : addressInstance.toString();
-  };
-  const fuelAddress = isValidAddress ? getAddress() : '';
+  const addressInstance = Address.fromDynamicInput(address);
```

Comment:
Maybe this is not that impactful performance-wise , but `fromDynamicInput` creates a new instance of `Address`, and without any memoization or state we'd be instantiating that for each render. 

I've also noticed we now need to do this inline on many places, maybe it'd be interesting to create a hook for this conversion that returns a memoized value?

Others might disagree here, I'm curious to know their opinion @LuizAsFight , @helciofranco, @pedronauck .

Reply:
yeah maybe a `useMemo` that only outputs the `b256` string would be more appropriate

Reply:
yeah maybe a useMemo that only outputs the b256 string would be more appropriate

Reply:
Great, nice suggestion!